apps/fz_http/lib/fz_http/int4range.ex

```diff
@@ -0,0 +1,70 @@
+defmodule FzHttp.Int4Range do
+  @moduledoc """
+  Ecto type for Postgres' Int4Range type
+  """
+  # Note: we represent a port range as a string: lower - upper for ease of use
+  # with Phoenix LiveView and nftables
+  use Ecto.Type
+
+  def type, do: :int4range
+
+  def cast(str) when is_binary(str) do
+    case parse_range(str) do
+      {:ok, range} -> cast(range)
+      err -> err
+    end
+  end
+
+  def cast([num, num]) when is_number(num) do
+    {:ok, Integer.to_string(num)}
+  end
+
+  def cast([lower, upper]) do
+    if upper >= lower do
+      {:ok, "#{lower} - #{upper}"}
+    else
+      {:error, message: "Range Error: Lower bound higher than upper bound"}
```

Comment:
If this is expected to happen should it be a validation instead?

Reply:
I tried adding it as a validation check but the moment you add an int4range to the DB where this happen it fails with an exception before the validation runs.